examples/bri-3/src/bri/transactions/api/transactions.controller.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { NotFoundException } from '@nestjs/common';
+import { CqrsModule } from '@nestjs/cqrs';
+import { Test, TestingModule } from '@nestjs/testing';
+import { NOT_FOUND_ERR_MESSAGE } from './err.messages';
+import { TransactionController } from './transactions.controller';
+import { TransactionAgent } from '../agents/transactions.agent';
+import { CreateTransactionCommandHandler } from '../capabilities/createTransaction/createTransactionCommand.handler';
+import { UpdateTransactionCommandHandler } from '../capabilities/updateTransaction/updateTransactionCommand.handler';
+import { DeleteTransactionCommandHandler } from '../capabilities/deleteTransaction/deleteTransactionCommand.handler';
+import { GetTransactionByIdQueryHandler } from '../capabilities/getTransactionById/getTransactionByIdQuery.handler';
+import { TransactionStorageAgent } from '../agents/transactionStorage.agent';
+import { CreateTransactionDto } from './dtos/request/createTransaction.dto';
+import { UpdateTransactionDto } from './dtos/request/updateTransaction.dto';
+import { MockTransactionStorageAgent } from '../agents/mockTransactionStorage.agent';
+
+describe('TransactionController', () => {
+  let controller: TransactionController;
+
+  beforeEach(async () => {
+    const app: TestingModule = await Test.createTestingModule({
+      imports: [CqrsModule],
+      controllers: [TransactionController],
+      providers: [
+        TransactionAgent,
+        CreateTransactionCommandHandler,
+        UpdateTransactionCommandHandler,
+        DeleteTransactionCommandHandler,
+        GetTransactionByIdQueryHandler,
+        TransactionStorageAgent,
+      ],
+    })
+      .overrideProvider(TransactionStorageAgent)
+      .useValue(new MockTransactionStorageAgent())
+      .compile();
+
+    controller = app.get<TransactionController>(TransactionController);
+
+    await app.init();
+  });
+
+  describe('getTransactionById', () => {
+    it('should throw NotFound if non existent id passed', () => {
+      // Arrange
+      const nonExistentId = '123';
+
+      // Act and assert
+      expect(async () => {
+        await controller.getTransactionById(nonExistentId);
+      }).rejects.toThrow(new NotFoundException(NOT_FOUND_ERR_MESSAGE));
+    });
+
+    it('should return the correct transaction if proper id passed ', async () => {
+      // Arrange
+      const requestDto = {
+        id: '123',
+        nonce: 1,
+        workflowInstanceId: '42',
+        workstepInstanceId: '24',
+        fromAccountId: 'from',
+        toAccountId: 'to',
+        payload: 'payload1',
+        signature: 'signature',
+      } as CreateTransactionDto;
+
+      const newTransactionId = await controller.createTransaction(requestDto);
+
+      // Act
+      const createdTransaction = await controller.getTransactionById(
+        newTransactionId,
+      );
+
+      // Assert
+      expect(createdTransaction.id).toEqual(requestDto.id);
```

Comment:
can we not just assert that two objects are equal?

Reply:
Could you please elaborate on what do you mean? They are not equal ATM because of the  TODOs on lines 81 and 82

Reply:
const expected = {from : '', to : '', ...requestDto}

expect(expected).toEqual(createdTransaction)

Reply:
I would rather keep it line by line because mappping of 1 to 1 from the requestDto to createdTransaction will not be held for long. Example is status field which is not provided in the dto but exits in the final result. Also update transaction will never use this approach since the dto only contains a subset of fields. In my eyes, it is better to go with a simple line by line assertions for the exact fields we are trying to check instead of trying to introduce generalizations that are not complete - that will have many exceptions. Not against generalizations but they should come at a later stage.

Regarding the _expect(expected).toEqual(actual)_ it is in fact the other way around, _expect(actual).toEqual(expected)_

Reply:
ok